torchvision/datasets/imagenet.py

```diff
@@ -66,49 +66,38 @@ def __init__(self, root, split='train', download=False, **kwargs):
                              for cls in clss}
 
     def download(self):
-        if not check_integrity(self.meta_file):
-            tmp_dir = tempfile.mkdtemp()
+        def check_archive(archive_dict):
+            archive = os.path.join(self.root, archive_dict["file"])
+            md5 = archive_dict["md5"]
+            if not check_integrity(archive, md5):
+                self._raise_download_error()
 
-            archive_dict = ARCHIVE_DICT['devkit']
-            download_and_extract_archive(archive_dict['url'], self.root,
-                                         extract_root=tmp_dir,
-                                         md5=archive_dict['md5'])
-            devkit_folder = _splitexts(os.path.basename(archive_dict['url']))[0]
-            meta = parse_devkit(os.path.join(tmp_dir, devkit_folder))
-            self._save_meta_file(*meta)
+            return archive
 
-            shutil.rmtree(tmp_dir)
+        if not check_integrity(self.meta_file):
+            archive = check_archive(ARCHIVE_DICT['devkit'])
+            parse_devkit(archive)
 
         if not os.path.isdir(self.split_folder):
-            archive_dict = ARCHIVE_DICT[self.split]
-            download_and_extract_archive(archive_dict['url'], self.root,
-                                         extract_root=self.split_folder,
-                                         md5=archive_dict['md5'])
+            archive = check_archive(ARCHIVE_DICT[self.split])
 
             if self.split == 'train':
-                prepare_train_folder(self.split_folder)
+                parse_train_archive(archive)
             elif self.split == 'val':
-                val_wnids = self._load_meta_file()[1]
-                prepare_val_folder(self.split_folder, val_wnids)
+                parse_val_archive(archive)
         else:
             msg = ("You set download=True, but a folder '{}' already exist in "
```

Comment:
Should this be converted into a warning instead of just printing it?

Reply:
Sure, a warning is fine